### ISSUE TITLE
How to discard packages on pause?

### ISSUE THREAD

**Issue by thefenix1** [muhku/FreeStreamer#135]
Thanks for the work on this, it's a really great component!

I'm consuming a remote stream and I need to resume after pausing on "current" content instead of playing previously buffered content. 

The behavior should be the same as a traditional radio: every time you turn it on you get the live stream and not a buffered one.

I need to use pause instead of stop since the stream plays an ad first time or after stop, we don't want the ad to play every single time.

So in short: I need to start fresh after pausing. Is there any way to do this? I'm willing to change the code if necessary but I'm having hard time to realize what change needs to be applied.

Thanks

**Comment by muhku** [comment]
The correct thing to do is to stop the stream if cached / buffered content is not an option. How long are the ads? You could do the following:
- Start playing the ad with FreeStreamer or any other player
- When the ad starts to play, set the radio stream volume to 0 and stop to the radio stream
- Depending on the length of the ad (let's guess it is about 10-15 secs), call preload to the radio stream when about half of the ad has been played
- When the ad has finished playing, call play and the radio stream starts playing immediately

The tricky part is to figure out how much buffering time you reserve when calling preload. If the buffering time is not enough, the user has to wait before the radio starts. If the buffering time is too long, the user will hear "cached" content. To be at the safe side, I would maybe start the playback earlier than too late because waiting is worse than a little more cached content.

Hope this helps.

**Comment by thefenix1** [comment]
Maybe I wasn't clear but the ad is part of the stream. Is sent on initial play or after stop. That's why I can't threat the ad as something different to the radio.

Thanks anyway, I appreciate you taking the time to answer. I wanted to hear from the owner of the component that there was no possible workaround using "pause".
